### PR TITLE
W-14998627: Mule Application plugin with unresolved property should not fail

### PR DESCRIPTION
**Pull request by aguozden**
[W-14998627](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001jkagJYAQ/view)

When building with MMP and having a project that is considered an application (see asApplication) and is also a mule-plugin (see Classifier), the final application will set some variables when using this plugin.
Therefore no error should be caught when building with MMP.
An integration test was implemented that reproduces the Bug issue.

New bug for Runtime, to implement fix: [W-15556985](https://gus.lightning.force.com/a07EE00001pKJUKYA4)
New user.story for MMP to remove temporary MMP fix when Runtime fix is implemented: [W-15554719](https://gus.lightning.force.com/a07EE00001pJKzwYAG)

Integrations test error before implementing MMP fix with if-else cause:
test result before MMP fix with try catch

Final Build :

### CODE REVIEW

**Review by egrave — CHANGES_REQUESTED** [review]

File: mule-extension-model-loader/src/main/java/org/mule/tooling/api/AstGenerator.java [hunk 1]
```diff
@@ -78,15 +80,22 @@ public AstGenerator(MavenClient mavenClient, String runtimeVersion,
...
       builder.withArtifactType(ArtifactType.DOMAIN);
     }
-    builder.withExtensionModels(extensionModels);
 
-    // Get a more specific error, so we can discriminate unresolved imports because the importFile has a property of because the
-    // file does not exist.
-    ConfigurationPropertiesResolver emptyPropertyResolver = new ConfigurationPropertiesHierarchyBuilder().build();
-    builder.withPropertyResolver(propertyKey -> (String) emptyPropertyResolver.resolveValue(propertyKey));
+    if (!asApplication || Classifier.MULE_PLUGIN.toString().equals(classifier)) {
+      emptyPropertyResolverBuilder.withoutFailuresIfPropertyNotPresent();
```
[the remark this answers is not in the document]
Comment (aguozden) [reply mule-extension-model-loader/src/main/java/org/mule/tooling/api/AstGenerator.java:92]: Thanks Edu, I have reorder the code.

**Review by WassimMelakhessou — APPROVED** [review]
LGTM, I suggest requesting more reviews.

**Review by egrave — DISMISSED** [review]

**Comment by aguozden** [comment]
!gate waiver add Sonar

**Comment by egrave** [comment]
!gate waiver add Sonar

**Review by egrave — APPROVED** [review]

**Pull request merged**